src/avif.c

```diff
@@ -264,11 +272,27 @@ void avifImageSetMetadataXMP(avifImage * image, const uint8_t * xmp, size_t xmpS
     avifRWDataSet(&image->xmp, xmp, xmpSize);
 }
 
-void avifImageAllocatePlanes(avifImage * image, avifPlanesFlags planes)
+avifResult avifImageAllocatePlanes(avifImage * image, avifPlanesFlags planes)
 {
-    int channelSize = avifImageUsesU16(image) ? 2 : 1;
-    int fullRowBytes = channelSize * image->width;
-    int fullSize = fullRowBytes * image->height;
+    if (image->width == 0 || image->height == 0) {
+        return AVIF_RESULT_INVALID_ARGUMENT;
+    }
+    if (image->depth != 8 && image->depth != 10 && image->depth != 12) {
+        return AVIF_RESULT_UNSUPPORTED_DEPTH;
+    }
+    const size_t channelSize = avifImageUsesU16(image) ? 2 : 1;
+    if (image->width > SIZE_MAX / channelSize) {
+        return AVIF_RESULT_INVALID_ARGUMENT;
+    }
+    const size_t fullRowBytes = channelSize * image->width;
+    if ((image->height > SIZE_MAX / fullRowBytes) || (fullRowBytes > UINT_MAX)) {
+        return AVIF_RESULT_INVALID_ARGUMENT;
+    }
+    const size_t fullSize = fullRowBytes * image->height;
```

Comment:
IMPORTANT: Also declare the `uvRowBytes` and `uvSize` variables (at lines 303-304) as `size_t`, similar to the `fullRowBytes` and `fullSize` variables here.